tests/integration/functions/types_test.ts

```diff
@@ -0,0 +1,162 @@
+            type: Schema.types.boolean,
+          },
+        },
+        obj: {
+          type: Schema.types.object,
+          properties: {
+            bool: {
+              type: Schema.types.boolean,
+            },
+          },
+          required: ["bool"],
+        },
+      },
+      required: ["bool", "int", "num", "string", "arr", "obj"],
+    },
+    output_parameters: {
+      properties: {
+        bool: {
+          type: Schema.types.boolean,
+        },
+        int: {
+          type: Schema.types.integer,
+        },
+        num: {
+          type: Schema.types.number,
+        },
+        string: {
+          type: Schema.types.string,
+        },
+        arr: {
+          type: Schema.types.array,
+          items: {
+            type: Schema.types.boolean,
+          },
+        },
+        obj: {
+          type: Schema.types.object,
+          properties: {
+            bool: {
+              type: Schema.types.boolean,
+            },
+          },
+          required: ["bool"],
+        },
+      },
+      required: ["bool", "int", "num", "string", "arr", "obj"],
+    },
+  });
+
+  type Actual = FunctionRuntimeType<typeof testFunctionDefinition>;
+
+  const expected: Actual = {
+    outputs: {
+      bool: true,
+      int: 0,
+      num: 0.1,
+      string: "",
+      arr: [true],
+      obj: {
+        bool: true,
+      },
+    },
+    args: {
+      inputs: {
+        bool: true,
+        int: 0,
+        num: 0.1,
+        string: "",
+        arr: [true],
+        obj: {
+          bool: true,
+        },
+      },
+      enterprise_id: "",
+      env: {},
+      team_id: "",
+      token: "",
+      client: SlackAPI("", {}),
+      event: {
+        event_ts: "",
+        function: {
+          id: "",
+          callback_id: "",
+          title: "",
+        },
+        function_execution_id: "",
+        type: "function_executed",
+      },
+    },
+  };
+
+  assert<CanBe<typeof expected, Actual>>(true);
+});
+
+Deno.test("FunctionRuntimeType should be able to provide a usable type of an empty DefineFunction return object", () => {
+  const testFunctionDefinition = DefineFunction({
+    callback_id: "test_function",
+    title: "Test function",
+    source_file: "functions/test_function.ts",
+    input_parameters: {
+      properties: {},
+      required: [],
+    },
+    output_parameters: {
+      properties: {},
+      required: [],
+    },
+  });
+
+  type Actual = FunctionRuntimeType<typeof testFunctionDefinition>;
+
+  const expected: Actual = {
+    outputs: {},
+    args: {
+      inputs: {},
+      enterprise_id: "",
+      env: {},
+      team_id: "",
+      token: "",
+      client: SlackAPI("", {}),
+      event: {
+        event_ts: "",
+        function: {
+          id: "",
+          callback_id: "",
+          title: "",
+        },
+        function_execution_id: "",
+        type: "function_executed",
+      },
+    },
+  };
+
+  assert<CanBe<typeof expected, Actual>>(true);
```

Comment:
Careful with the order of generics/type parameters to `CanBe`! `CanBe<T, undefined>` is much different than `CanBe<undefined, T>`! I think about this as set membership: `CanBe<T, U>` translates to "is `U` included in the set of `T`?", or also "can `T` be a `U`?" The implication being that, if the assertion is true, we expect that T is a superset of U.

Possibly we may want to clarify this at the `CanBe` level under `src/test_utils.ts` via documentation and/or better naming of the type parameters; `T` and `U` are pretty generic and unclear. Or maybe even renaming the utility type to make it clear! Open to suggestions.